File: APCS/2021-11/p1-g595.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int n; cin >> n;
    int h[101], ans = 0;;
    for(int i = 0; i < n; i++) cin >> h[i];
    for(int i = 0; i < n; i++)
    {
        if(h[i] == 0)
        {
            if(i == 0) ans += h[i+1];
            else if(i == n-1) ans += h[n-2];
            else ans += min(h[i-1], h[i+1]);
        }
    }
    cout << ans << '\n';
}

File: APCS/2021-11/P1-g595.py
n = int(input())
h = list(map(int,input().split(' ')))
ans = 0
for i in range(n):
    if h[i] == 0:
        if i == 0:
            ans += h[i+1]
        elif i == n-1:
            ans += h[n-2]
        else:
            ans += min(h[i-1],h[i+1])
print(ans)